src/auth.js

```diff
@@ -136,7 +136,7 @@ export async function authorizeEntry(
   } else {
     signature = Buffer.from(signer.sign(payload));
   }
-  const publicKey = Address.fromScAddress(addrAuth.address()).toString();
+  const publicKey = signer.publicKey();
```

Comment:
`signer` can be a `function` (see previous `if` statement) if you're passing a callback; this will only work in the `Keypair` case (hence test failures)